test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,275 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+		framework.ExpectNoError(err)
+
+		serverPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-server",
+		}
+
+		// Step 1: set up iperf2 server -- a single pod on any node
+		serverDeployment, err := iperf2ServerDeployment(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server deployment")
+
+		_, err = iperf2ServerService(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 server service")
+
+		// Step 2: set up iperf2 client daemonset
+		//   initially, the clients don't do anything -- they simply pause until they're called
+		_, err = iperf2ClientDaemonSet(f.ClientSet, f.Namespace.Name)
+		framework.ExpectNoError(err, "deploy iperf2 client daemonset")
+
+		clientPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-client",
+		}
+
+		// Make sure the server is ready to go
+		framework.ExpectNoError(e2edeployment.WaitForDeploymentComplete(f.ClientSet, serverDeployment), "unable to wait for server deployment")
+
+		// Make sure the clients are ready to go
+		err = wait.PollImmediate(2*time.Second, 100*time.Second, func() (done bool, err error) {
+			clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+			if err != nil {
+				return false, err
+			}
+			if len(clientPodList.Items) < len(nodeList.Items) {
+				return false, nil
+			}
+			for _, pod := range clientPodList.Items {
+				if pod.Status.Phase != v1.PodRunning {
+					return false, nil
+				}
+			}
+			return true, nil
+		})
+		framework.ExpectNoError(err, "unable to wait for client pods to come up")
+
+		// Get a reference to the server pod for later
+		serverPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), serverPodsListOptions)
+		framework.ExpectNoError(err)
+		if len(serverPodList.Items) != 1 {
+			framework.Failf("expected 1 server pod, found %d", len(serverPodList.Items))
+		}
+		serverPod := serverPodList.Items[0]
+
+		// Get a reference to the client pods for later
+		clientPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(context.TODO(), clientPodsListOptions)
+		framework.ExpectNoError(err)
+		framework.Logf("found %d matching client pods", len(clientPodList.Items))
+		if len(clientPodList.Items) == 0 {
+			framework.Failf("found 0 matching client pods")
+		}
+
+		// Keep track of all the iperf2 results
+		nodeResults := &IPerf2NodeToNodeResults{
+			ServerNode: serverPod.Spec.NodeName,
+			Results:    map[string]*IPerf2EnhancedResults{},
+		}
+		// Step 3: iterate through the client pods one by one, running iperf2 in client mode to transfer
+		//   data to the server and back and measure bandwidth
+		for _, pod := range clientPodList.Items {
+			podName := pod.Name
+			nodeName := pod.Spec.NodeName
+			hostIP := pod.Status.HostIP
+
+			command := fmt.Sprintf(`iperf -e -p %d --reportstyle C -i 1 -c iperf2-server -T "Client on %s"`, iperf2Port, hostIP)
```

Comment:
add a comment explaining the iperf flags please
// iperf
// -e enhanced report
// -p iperf2Port destination port
// -reportstyle C ...
// -i 1 report every 1 second interval
// -c iperf2-server connect to the iperf2-server service
..

Reply:
👍